### DISCUSSION TITLE
Considerations for using PVE

### DISCUSSION THREAD

**Discussion by BOBINIUNIU** [daeuniverse/dae#37] [General]
adguardhome and dae are finally working properly! I will share some experiences.

1. If your environment is PVE, don't use the hardware passthrough feature, it may cause some weird issues
2. Important! Set the virtual machine's NIC to Intel E1000 instead of Virtio
3. The part of the documentation for DNS configuration is wrong, please update it to the following configuration
```
dns {
  upstream {
    adguardhome: 'udp://127.0.0.1:53'
  }
  routing {
    request {
      fallback: adguardhome
    }
  }
}
```

My system environment: NIC: Intel i226, PVE: 7.3-6/Kernel 6.1.15-1, VM: Debian 11.6

**Comment by mzz2017** [comment]
Thank you! I'll update it later.

**Reply by mzz2017** [reply]
> 3. The part of the documentation for DNS configuration is wrong, please update it to the following configuration

I have updated the External DNS docs. Thanks. @BOBINIUNIU

**Comment by BOBINIUNIU** [comment]
Should we need change the following lines to must_direct for avoid traffic going through DNS?

pname(NetworkManager, systemd-resolved) -> direct
dip(224.0.0.0/3, 'ff00::/8') -> direct

**Reply by sontjer** [reply]
> Should we need change the following lines to must_direct for avoid traffic going through DNS?
> 
> pname(NetworkManager, systemd-resolved) -> direct dip(224.0.0.0/3, 'ff00::/8') -> direct

sip(192.168.x.x) && l4proto(udp) && dport(53) -> must_direct 
Should this line above be inserted on top of routing section?

Thanks for the heads-up on external DNS configuration! Me, on the same version and kernel of PVE as well, but I can run the latest version of dae on VM debian 11 smoothly as butter with the same specification as yours, apart from NIC intel i226, mine is i 210.

**Reply by mzz2017** [reply]
> sip(192.168.x.x) && l4proto(udp) && dport(53) -> must_direct Should this line above be inserted on top of routing section?

@sontjer Yes. You should put it top if your external DNS is on another machine in LAN.

**Comment by mzz2017** [comment]
@BOBINIUNIU It is not necessary if dae's upstreams not contain them.

**Comment by Pacio** [comment]
hi, May I set the virtual machine's NIC to VMXNET3? 
cause I don’t want to limit the internet speed to 1000m
